frontend/src/general/SnackbarMessage.jsx

```diff
@@ -0,0 +1,166 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Snackbar, SnackbarContent, IconButton, Box } from "@material-ui/core";
+import { CheckCircle, Error, Close } from "@material-ui/icons";
+import { withRouter } from "react-router";
+
+let openSnackbarFn;
+
+class SnackbarMessage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      isSuccess: false,
+      status: "",
+      message: "",
+      pathname: ""
+    };
+    this.setOpen = this.setOpen.bind(this);
+    this.setIsSuccess = this.setIsSuccess.bind(this);
+    this.setStatus = this.setStatus.bind(this);
+    this.setMessage = this.setMessage.bind(this);
+    this.setPathname = this.setPathname.bind(this);
+    this.addSnackbarProperties = this.addSnackbarProperties.bind(this);
+    this.openSnackbar = this.openSnackbar.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.refresh = this.refresh.bind(this);
+  }
+
+  componentDidMount() {
+    openSnackbarFn = this.openSnackbar;
+  }
+
+  handleClose(event, reason) {
+    if (reason === "clickaway") {
+      return;
+    }
+    this.setOpen(false);
+  }
+
+  setOpen(open) {
+    this.setState({
+      open
+    });
+  }
+
+  setStatus(status) {
+    this.setState({
+      status
+    });
+  }
+
+  setIsSuccess(isSuccess) {
+    this.setState({
+      isSuccess
+    });
+  }
+
+  setMessage(message) {
+    this.setState({
+      message
+    });
+  }
+
+  setPathname(pathname) {
+    this.setState({
+      pathname
+    });
+    return pathname;
+  }
+
+  addSnackbarProperties(stat, msg) {
+    this.setStatus(stat);
+    if (stat !== "error") {
+      this.setIsSuccess(true);
+    } else {
+      this.setIsSuccess(false);
+    }
+    this.setMessage(msg);
+  }
+
+  openSnackbar(stat, msg, path) {
+    const currentPathname = this.setPathname(path);
+    if (currentPathname !== "") {
+      this.refresh();
+      setTimeout(() => {
+        this.addSnackbarProperties(stat, msg);
+        this.setOpen(true);
+      }, 300);
+    } else {
+      this.addSnackbarProperties(stat, msg);
+      this.setOpen(true);
+    }
+  }
+
+  refresh() {
+    const { history, location } = this.props;
+    const { pathname } = this.state;
+    if (location.pathname.endsWith(pathname)) {
+      history.go(0);
+    } else {
+      history.push(`/${pathname}`);
+    }
+  }
+
+  render() {
+    const { open, isSuccess, status, message } = this.state;
+    return (
+      <>
+        <Snackbar
+          contentprops={{
+            "aria-describedby": "message-id"
+          }}
+          anchorOrigin={{
+            vertical: "top",
+            horizontal: "right"
+          }}
+          open={open}
+          autoHideDuration={3000}
+          onClose={this.handleClose}
+        >
+          <SnackbarContent
+            style={{
+              backgroundColor: isSuccess ? "#4caf50" : "#f44336"
+            }}
+            // prettier-ignore
```

Comment:
Why `prettier-ignore`?

Reply:
This is because the linter will make the following couple lines with the conditional rendering not very "human-readable" with how they indent and space out the next couple of lines, but I can remove it if that's an issue.

Reply:
I don't really mind that much